libvast/src/system/exporter.cpp

```diff
@@ -89,18 +92,28 @@ void request_more_hits(stateful_actor<exporter_state>* self) {
 } // namespace <anonymous>
 
 behavior exporter(stateful_actor<exporter_state>* self, expression expr,
-                  query_options) {
+                  query_options opts) {
   auto eu = self->system().dummy_execution_unit();
   self->state.sink = actor_pool::make(eu, actor_pool::broadcast());
   if (auto a = self->system().registry().get(accountant_atom::value))
     self->state.accountant = actor_cast<accountant_type>(a);
+  self->state.opts = opts;
+  if (has_continuous_option(opts))
+    VAST_DEBUG(self, "has continuous query option");
   self->set_exit_handler(
     [=](const exit_msg& msg) {
       self->send(self->state.sink, sys_atom::value, delete_atom::value);
       self->send_exit(self->state.sink, msg.reason);
       self->quit(msg.reason);
     }
   );
+  self->set_down_handler(
+    [=](const down_msg& msg) {
+      VAST_DEBUG(self, "received DOWN from", msg.source);
+      if (msg.reason == exit_reason::user_shutdown)
+        shutdown(self);
```

Comment:
> What I mean is that the process that receives the CTRL + C might be a different process than the process that hosts the EXPORTER, right?

Ah, right. That can be the case. [When sinks get registered](https://github.com/vast-io/vast/blob/9a0688897baedc292a6a95718e57337b72a367b5/libvast/src/system/exporter.cpp#L213), they must be monitored then. Then, with the `down_msg` logic described above, we should be able to catch when a SINK is gone. This brings up one problem though: right now is that the SINKs are hidden behind an actor pool, and if one SINK goes down, the EXPORTER should still be running. Only if the last sink disconnects, the EXPORTER can shutdown. Maybe we should simply monitor the pool of sinks? IIRC it will shutdown itself if it runs out of actors.